### PR TITLE
Support for Numeric type for row filter

### PR DESCRIPTION
**Pull request by CuriousVini**
Issue: https://issues.cask.co/browse/CDAP-12428

### CODE REVIEW

File: wrangler-core/src/main/java/co/cask/directives/row/RecordRegexFilter.java [hunk 1]
```diff
@@ -105,13 +101,11 @@ public void destroy() {
...
-            matches = !matches;
-          }
-          if (matches) {
-            continue;
-          }
+          if (isMatching(value)) continue;
+        } else if (object instanceof Number) {
+          Number number = (Number) row.getValue(idx);
+          String value = number.toString();
+          if (isMatching(value)) continue;
```
Comment (bdmogal) [inline wrangler-core/src/main/java/co/cask/directives/row/RecordRegexFilter.java]: break this for consistency?

**Review by bdmogal — APPROVED** [review]
lgtm, but can you get another set of eyes on this?

**Comment by nitinmotgi** [comment]
Don't see test cases. Please add test cases for int,long,short,string,double,float,Integer,Short,Long,Double,Float please.

File: wrangler-core/src/main/java/co/cask/directives/row/RecordRegexFilter.java [hunk 2]
```diff
@@ -105,13 +101,11 @@ public void destroy() {
...
         } else if (object instanceof String) {
           String value = (String) row.getValue(idx);
-          boolean matches = pattern.matcher(value).matches(); // pattern.matcher(value).matches();
-          if(!matched) {
-            matches = !matches;
-          }
-          if (matches) {
-            continue;
-          }
+          if (isMatching(value)) continue;
```
Comment (nitinmotgi) [inline wrangler-core/src/main/java/co/cask/directives/row/RecordRegexFilter.java]: put { } before and after continue.

Comment (nitinmotgi) [inline wrangler-core/src/main/java/co/cask/directives/row/RecordRegexFilter.java]: Optimize code were you put isMatching in only one place. If so, then you don't need isMatching function.

**Comment by CuriousVini** [comment]
@nitinmotgi Thanks for the review. I have added the test and optimized the code. However, I could only add test for Integer, Double and String types because of https://issues.cask.co/browse/CDAP-12605.

**Comment by CuriousVini** [comment]
Thanks @nitinmotgi I have added the tests and optimized the code. Please take a quick look when you get a chance, thanks

**Comment by nitinmotgi** [comment]
LGTM

**Pull request merged**